### PR TITLE
PLT-3610 Adding migration support to LDAP from other account types

### PR DESCRIPTION
**Pull request by crspeller**
#### Summary
Adds command to migrate users to LDAP accounts. 

Related Enterprise PR: https://github.com/mattermost/enterprise/pull/45

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Has enterprise changes (please link)
- [x] Includes text changes and localization files updated

### CODE REVIEW

**Comment by jwilander** [comment]
I imagine this is failing because the EE PR needs to be merged first?

**Comment by crspeller** [comment]
Nope, the build server handles that. Something strange is happening. Working on it.

**Comment by crspeller** [comment]
@jwilander Fixed!

File: mattermost.go [hunk 1]
```diff
@@ -1475,6 +1485,44 @@ func cmdRunLdapSync() {
...
 
+func cmdRunMigrateAccounts() {
+	if flagCmdMigrateAccounts {
+		if len(flagFromAuth) == 0 || (flagFromAuth != "email" && flagFromAuth != "gitlab" && flagFromAuth != "saml") {
+			fmt.Fprintln(os.Stderr, "flag needs an argument: -from_auth")
+			flag.Usage()
+			os.Exit(1)
+		}
+
+		if len(flagToAuth) == 0 || flagToAuth != "ldap" {
```
Comment (coreyhulen) [inline mattermost.go]: what about supporting LDAP to Email?

Comment (crspeller) [inline mattermost.go]: Separate ticket.

**Comment by enahum** [comment]
+1

**Comment by coreyhulen** [comment]
+1

**Pull request merged**